kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -71,7 +72,17 @@ public CompletionStage<RequestFilterResult> onProduceRequest(short apiVersion,
                                                                  ProduceRequestData request,
                                                                  FilterContext context) {
         return maybeEncodeProduce(request, context)
-                .thenCompose(yy -> context.forwardRequest(header, request));
+                .thenCompose(yy -> context.forwardRequest(header, request))
+                .exceptionallyCompose(throwable -> {
+                    if (throwable instanceof RequestNotSatisfiable || throwable.getCause() instanceof RequestNotSatisfiable) {
+                        return context.requestFilterResultBuilder()
+                                .errorResponse(header, request, throwable)
+                                .completed();
+                    }
+                    else {
+                        return CompletableFuture.failedStage(throwable);
```

Comment:
Perhaps comment that the intended effect of the `failedStage` is disconnection?

Reply:
Could do, but thinking about it, maybe it should be handled as 
```suggestion
                        return context.requestFilterResultBuilder()
                                .errorResponse(header, request, throwable)
                                .completed();
                    }
                    else {
                         return context.requestFilterResultBuilder()
                                .drop()
                                .completed();

```

Though that would lose the logging and context. So perhaps better to let it propagate with a comment warning that it will kill the connection.

Reply:
I added a comment in https://github.com/kroxylicious/kroxylicious/pull/1630/commits/f1939468381da9cb119f5ac9230c411081292981 

The suggestion could still be worth doing in another PR